### PR TITLE
Green 86 ajv first endpoint validator repo

### PR DESCRIPTION
**Pull request by tonititi**
Add the AJV verification for the 'inject' endpoint of validator

### CODE REVIEW

File: src/types/InjectReq.ts
Comment (arhamj) [inline src/types/InjectReq.ts:1]: Why do we need these ser/de methods? I don't see them being used.

File: src/types/InjectResp.ts
Comment (arhamj) [inline src/types/InjectResp.ts:1]: ser/de methods not needed

File: src/types/ajv/InjectResp.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { addSchema } from '../../utils/serialization/SchemaHelpers'
+export const schemaInjectResponse = {
```
Comment (arhamj) [inline src/types/ajv/InjectResp.ts:3]: Add unit test to ensure this schema works as expected

File: src/types/ajv/InjectReq.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { addSchema } from '../../utils/serialization/SchemaHelpers'
+export const schemaInjectReq = {
```
Comment (arhamj) [inline src/types/ajv/InjectReq.ts:3]: Add unit test to ensure this schema works as expected

**Review by arhamj — CHANGES_REQUESTED** [review]

File: src/utils/serialization/VectorBufferStream.ts [hunk 3]
```diff
@@ -0,0 +1,223 @@
+export class VectorBufferStream {
+    private buffer: Buffer
```
Comment (S0naliThakur) [inline src/utils/serialization/VectorBufferStream.ts:2]: Use VectorBufferStream from shardus core. Request you to test out the chenges in a local nw

File: src/types/ajv/InjectReq.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  addSchemas()
+}
+// Function to add schema dependencies
+function addSchemaDependencies(): void {
+  // No dependencies
+}
+
+// Function to register the schema
+function addSchemas(): void {
+  addSchema('InjectReq', schemaInjectReq)
```
Comment (S0naliThakur) [inline src/types/ajv/InjectReq.ts]: Can consider creating enum for schema names 'InjectReq' and 'InjectResp'

**Review by S0naliThakur — CHANGES_REQUESTED** [review]

**Comment by S0naliThakur** [comment]
Also remove extra comments from the tests

File: src/utils/serialization/VectorBufferStream.ts (hunk 3 above)
Comment (S0naliThakur) [inline src/utils/serialization/VectorBufferStream.ts:2]: Use VectorBufferStream from shardus core. Request you to test out the chenges in a local nw

**Review by S0naliThakur — CHANGES_REQUESTED** [review]

File: src/utils/serialization/VectorBufferStream.ts [hunk 5]
```diff
@@ -0,0 +1,223 @@
+export class VectorBufferStream {
```
Comment (S0naliThakur) [inline src/utils/serialization/VectorBufferStream.ts:1]: Unused code. Remove this

File: package.json [hunk 6]
```diff
@@ -29,7 +29,8 @@
     "build-docker-production": "docker build -f dev.Dockerfile -t registry.gitlab.com/shardeum/server .",
     "push-docker-production": "docker push registry.gitlab.com/shardeum/server",
     "tsc": "tsc",
-    "dcr": "node scripts/debugCycleRecords.ts"
+    "dcr": "node scripts/debugCycleRecords.ts",
+    "test-ajv": "jest"
```
Comment (S0naliThakur) [inline package.json:33]: can reuse test only

File: test/unit/src/types/InjectReq.test.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                },
+                timestamp: 2,
+                accountId: '0x1',
+            },
+        ]
+        test.each(invalidObjects)('should throw AJV error', (data) => {
+            const res = verifyPayload('InjectReq', {
+                ...data,
+            })
+            console.log('res', res!)
```
Comment (S0naliThakur) [inline test/unit/src/types/InjectReq.test.ts:43]: remove extra console logs, unnecessary comments from all the unit tests. Some tests fail, fix them too

**Review by S0naliThakur — CHANGES_REQUESTED** [review]

**Pull request closed without merging**